internal/cluster/transition.go

```diff
@@ -524,6 +525,56 @@ func (th *transitionHandler) PostPreparingTimedOut(sw stateswitch.StateSwitch, a
 		updatedCluster *common.Cluster
 	)
 
+	// Capture some information about hosts states, we should be able to determine more about the cause of the timeout.
+	// This can occur because the host...
+	//
+	// 1: Becomes disconnected - the "connected" validation of the host will be failing
+	// 2: The media has been disconnected - The "media connected" validation of the host will be failing
+	// 3: The condition "installation-disk-speed-check-succesful" is false
+	// 4: The condition "succeful-container-image-availability" is false
+	//
+	issues := []string{}
+	
+	for _, clusterHost := range sCluster.cluster.Hosts {
+		if clusterHost.ValidationsInfo != "" {
+			// Deserialize validations info
+			var hostValidations ValidationsStatus
+			err := json.Unmarshal([]byte(clusterHost.ValidationsInfo), &hostValidations)
+			if err != nil {
+				issues = append(issues, fmt.Sprintf("unable to deserialize validations info for host %s due to error %s", clusterHost.ID, err.Error()))
+			}
+			for _, vRes := range hostValidations {
+				for _, v := range vRes {
+					if v.Status == ValidationFailure {
+						issues = append(issues, fmt.Sprintf("validation with ID %s is failing for host %s", models.ClusterValidationID(v.ID), clusterHost.ID))
+					}
+				}
+			}
+		}
+
+		if clusterHost.Status != nil && *clusterHost.Status == models.HostStatusInsufficient {
+			// Indicates that the host has failed the disk speed check
+			issues = append(issues, fmt.Sprintf("host %s has failed the installation disk speed check", clusterHost.ID))
+		}
+
+		if clusterHost.Status != nil && *clusterHost.Status == models.HostStatusPreparingFailed {
+			// Indicates that the host has failed the disk speed check
+			issues = append(issues, fmt.Sprintf("host %s has failed to fetch images", clusterHost.ID))
```

Comment:
We already have this information from the container-images-available validation.